engine/include/Engine/Intermediate/GameObject.hpp

```diff
@@ -0,0 +1,209 @@
+     *
+     * @tparam T
+     * @return Component&
+     */
+    template <typename T>
+    [[nodiscard]] inline T* getComponent() noexcept;
+
+    /**
+     * @brief Get the first gameObject with path in arg
+     *
+     * @param path : example world/car/motor/piston3 or car/motor/piston3 or ./car/motor/piston3
+     * @return GraphEntity&
+     */
+    [[nodiscard]] GameObject* getChild(const std::string& path) noexcept;
+
+    /**
+     * @brief Destroye the first gameObject with path in arg
+     *
+     * @param path : example world/car/motor/piston3 or car/motor/piston3 or ./car/motor/piston3
+     * @return GraphEntity&
+     */
+    void destroyChild(const std::string& path) noexcept; // TODO: Destroy immediate may be dangerous
+
+    /**
+     * @brief destroy childen of gameobject
+     *
+     * @param GameObject
+     */
+    std::list<std::unique_ptr<GameObject>>::iterator destroyChild(
+        GameObject* pGameObject) noexcept; // TODO: Destroy immediate may be dangerous
+
+    inline std::list<std::unique_ptr<GameObject>>::iterator destroyChild(
+        const std::list<std::unique_ptr<GameObject>>::iterator&
+            it) noexcept; // TODO: Destroy immediate may be dangerous
+
+    template <typename TUniqueComponentType>
+    void destroyImmediateUniqueComponent() noexcept;
+
+    /**
+     * @brief destroy component of gameobject
+     *
+     * @param Component
+     */
+    std::list<Component*>::iterator destroyComponent(
+        Component* pComponent) noexcept; // TODO: Destroy immediate may be dangerous
+
+    inline void setActive(bool newState);
+
+    inline std::list<Component*>::iterator destroyComponent(
+        const std::list<Component*>::iterator& it) noexcept; // TODO: Destroy immediate may be dangerous
+
+    /**
+     * @brief Destroy the element at the next frame whe scene graph is update.
+     *
+     */
+    inline void destroy() noexcept;
+
+    /**
+     * @brief Destroy the element now, don't way the next frame.
+     * WARNING : if element his destroy during update of system, it can create conflict or segfault.
+     * Use this function only if you are sur that the object don't depending af the current frame
+     *
+     */
+    inline void destroyImmediate() noexcept;
+
+    /**
+     * @brief add specific entity to the graph with arg to construct it and return his id
+     *
+     * @tparam T
+     * @tparam Args
+     * @param args
+     * @param dependenceEntity&
+     * @return GameObject&
+     */
+    template <typename T, typename... Args>
+    GameObject& addChild(Args&&... args) noexcept;
+
+    [[nodiscard]] inline constexpr bool operator==(GameObject const& other) noexcept;
+
+    template <typename T>
+    [[nodiscard]] std::vector<T*> getComponents() noexcept;
+
+    [[nodiscard]] inline constexpr std::list<Component*>& getComponents() noexcept;
+
+    [[nodiscard]] inline constexpr const std::list<Component*>& getComponents() const noexcept;
```

Comment:
**Suggestion**

Make `m_Components` public

Reply:
maintenability

Reply:
inline constexpr std::list<Component*>& getComponents() noexcept; is not safe and must probabely change